### DISCUSSION TITLE
Config system

### DISCUSSION THREAD

**Discussion by Jaccodouma** [EveronLife/EveronLife#67] [Ideas]
I think the framework should provide a system for config files. Probably have separate ones for Server Config and Player Config (preferences). Would keep all modules from having to implement some config file logic themselves.

For example ServerConfig.getConfig(name, structure) could get a config and create the file with the right structure and default values if it doesn't exist.

**Comment by Arkensor** [comment]
The question is if we really want to do configs via file. Server owners will need to set up their world anyway where they can simply toggle features by not having things on the map or ticking checkboxes on components or choosing dropdowns there. There are lots of things we can not change at runtime so server owners will need to go into the world editor to manage their server anyway. So we might want to do all there. We can always provide a quick configuration wizard as a workbench plugin in there so they don't need to look where to change the options for too long.

**Reply by Jaccodouma** [reply]
True! Hadn't thought of that. I imagine a player preferences handler would still be relevant though.
